src/cow-react/modules/trade/docs/README.md

```diff
@@ -0,0 +1,38 @@
+# Trade module
+
+## Problem statement
+
+Since we have at least three trade widgets we need to unify their logic and avoid code duplication  ([DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)). To do this, we need to define what is specific and what is common.
+At the same time, strike a balance between code cohesion and solution scalability.
+
+## Shared language
+
+- `TradeWidget` - Widget to create CoW orders. Its configuration can create many different ways to create orders, for example Swap, Limit Orders and Advanced orders will be instances of it
+- `TradeRawState` - DTO derived from URL (`/1/USDT/WETH?recipient=0x000`) and persisted into localStorage ([ref](../../../modules/trade/types/TradeRawState.ts))
+- `TradeDerivedState` - a model that contains `CurrencyAmount` and `Currency` instances. This state is derived from TradeRawState ([ref](../../../modules/trade/types/TradeDerivedState.ts))
+
+## Concept of TradeWidget
+
+1. Each trade widget: [SwapWidget](../../../modules/swap/containers/SwapWidget/index.tsx), [LimitOrdersWidget](../../../modules/limitOrders/containers/LimitOrdersWidget/index.tsx), [AdvancedOrdersWidget](../../../modules/advancedOrders/containers/AdvancedOrdersWidget/index.tsx) uses [TradeWidget](../../../modules/trade/containers/TradeWidget/index.tsx) as a skeleton to get out of the box:
```

Comment:
For the chart i have some comments, but I think is good enough to leave it like this for now. 

I specially like the trade widget extenssion